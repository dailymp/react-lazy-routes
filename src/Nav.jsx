import React from 'react';
import { Link } from './routing/index.jsx'; // TODO: configure node-resolve

const getLinkText = path =>
  `${path[1].toUpperCase()}${path.slice(2)}`;

const Nav = ({ paths }) => (
  <nav>
    <ul>
      {paths.map((path, i) => (
        <li key={i}>
          <Link href={path}>
            {getLinkText(path)}
          </Link>
        </li>
      ))}
    </ul>
  </nav>
);

export default Nav;
