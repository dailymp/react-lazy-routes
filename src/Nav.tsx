import * as React from 'react';
import { Link } from './routing';

interface NavProps {
  paths: string[];
}

const getLinkText = (path: string) =>
  `${path[1].toUpperCase()}${path.slice(2)}`;

const Nav: React.FC<NavProps> = ({ paths }) => (
  <nav>
    <ul>
      {paths.map(path => (
        <li key={path}>
          <Link href={path}>{getLinkText(path)}</Link>
        </li>
      ))}
    </ul>
  </nav>
);

export default Nav;
