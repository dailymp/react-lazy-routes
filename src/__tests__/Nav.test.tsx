import * as React from 'react';
import * as testRenderer from 'react-test-renderer';
import Nav from '../Nav';

describe('Nav', () => {
  it('should map an array of paths into a list of links, computing the text of each', () => {
    const paths = ['/foo', '/bar', '/baz'];
    const rendered = testRenderer.create(<Nav paths={paths} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
