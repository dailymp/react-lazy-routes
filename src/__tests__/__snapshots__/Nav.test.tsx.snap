// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should map an array of paths into a list of links, computing the text of each 1`] = `
<nav>
  <ul>
    <li>
      <a
        href="/foo"
        onClick={[Function]}
      >
        Foo
      </a>
    </li>
    <li>
      <a
        href="/bar"
        onClick={[Function]}
      >
        Bar
      </a>
    </li>
    <li>
      <a
        href="/baz"
        onClick={[Function]}
      >
        Baz
      </a>
    </li>
  </ul>
</nav>
`;
